crates/iota-sdk-types/src/crypto/zklogin.rs

```diff
@@ -143,6 +165,20 @@ pub struct CircomG1(pub [Bn254FieldElement; 3]);
 #[cfg_attr(feature = "proptest", derive(test_strategy::Arbitrary))]
 pub struct CircomG2(pub [[Bn254FieldElement; 2]; 3]);
 
+#[cfg(feature = "uniffi")]
+uniffi::custom_type!(CircomG2, Vec<Vec<Bn254FieldElement>>, {
+    lower: |c| c.0.into_iter().map(|v| v.to_vec()).collect(),
+    try_lift: |v| Ok(CircomG2(
+        v.into_iter()
+            .map(|v|
+                <[Bn254FieldElement; 2]>::try_from(v)
+                    .map_err(|_| anyhow!("failed to convert field elements to CircomG2"))
+            ).collect::<Result<Vec<_>, _>>()?
+            .try_into()
+            .map_err(|_| anyhow!("failed to convert field elements to CircomG1"))?
```

Comment:
Actually the first message here should mention `Bn254FieldElement` and the second should be something about not being able to convert the vec to an array I think